src/query/arithmetic.c

```diff
@@ -5242,25 +5241,23 @@ db_evaluate_json_length (DB_VALUE * result, DB_VALUE * const *arg, int const num
   if (path != NULL)
     {
       const char *raw_path = db_get_string (path);
-      JSON_DOC *extracted_doc = NULL;
+      JSON_DOC_STORE extracted_doc;
 
       /* *INDENT-OFF* */
-      error_code = db_json_extract_document_from_path (source_doc, std::vector<std::string> (1, raw_path), extracted_doc, false);
+      error_code = db_json_extract_document_from_path (source_doc.get_immutable_reference (), {raw_path}, extracted_doc, false);
```

Comment:
Why not use `source` as output for db_json_extract_document_from_path?

Indent split line

Reply:
I really don't like it because then any call during db_json_extract_document_from_path of result.set_as_mutable_reference() would destroy the first document argument, I think it's very dangereous